tests/ml/test_ml_model_pytest.py

```diff
@@ -438,6 +448,7 @@ def test_learning_to_rank(self, objective, compress_model_definition):
             pass
 
         # Clean up
+        ES_TEST_CLIENT.cluster.health(index='.ml-*', wait_for_active_shards='all')
```

Comment:
I did add this to prevent test flakiness.

Indeed, the the deletion of the model sometimes failed because of a search_phase exception. Under the hood it is caused by the cleaning of the stats happening during deletion that is occurring before the ml-stats index has all his shards assigned.

This is probably something to be fixed in ES (and clearly not related to this PR).